### DISCUSSION TITLE
Setting Max Parallelism on Refresh

### DISCUSSION THREAD

**Discussion by EthanEverett** [TabularEditor/TabularEditor3#269] [🙋♀️TE3 Q&A]
For some models the data sources get overwhelmed due to the number of concurrent requests and will time out. We are able to control for this by deploying to SSAS and then specifying Max Parallelism when refreshing in SSMS via the code below:

{
"sequence": {
"maxParallelism": 5,
"operations": [
{
"refresh": {
"type": "full",
"objects": [
{
"database": "TabularModel"
}
]
}
}
]
}
}

Is this the same as the Max Connections setting? Or is there a setting available for specifying Max Parallelism in TE? If so, any help is much appreciated.

**Comment by helgews** [comment]
alwortman answered:

Configurable on a per-datasource basis via "Max Connections" property. That is the easiest way to prevent these kinds of errors.

![image](https://user-images.githubusercontent.com/56156227/137580171-aacfb1ab-3fef-49c7-b24d-8347c59780c9.png)

https://docs.microsoft.com/en-us/analysis-services/multidimensional-models/set-data-source-properties-ssas-multidimensional?view=asallproducts-allversions

I would be curious how TE3 generates the XMLA sequence command when refreshing multiple tables at once...maybe there is something there?

https://docs.microsoft.com/en-us/analysis-services/tmsl/sequence-command-tmsl?view=asallproducts-allversions

**Comment by helgews** [comment]
I guess you could monitor the model using sql server profiler or Azure Log Analytics, to capture the refresh commands? 
I feel like If I select multiple tables and chose refresh, it puts them in sequence. If I chose model and refresh, it uses paralelism like @EthanEverett posted about.
